sdk/identity/azure-identity/test/ut/azure_pipelines_credential_test.cpp

```diff
@@ -151,7 +151,7 @@ TEST(AzurePipelinesCredential, InvalidArgs)
   }
 }
 
-TEST(AzurePipelinesCredential, Regular)
+TEST(AzurePipelinesCredential, DISABLED_Regular)
```

Comment:
The implementation works as expected given the live test `RegularLive` passes.

This test is simulating a response using `SimulateTokenRequest` but that test helper infra isn't able to correctly simulate returning a response when the credential type under the covers is different than the one the test helper was constructed with. We need to update the test helper to continue using the TestTransport to return the mocked response, whenever GetToken is called.

Right now, the CredentialTestHelper ends up sending the test request to the actual endpoint and fails to return the mocked response, which causes the test to fail.

cc @antkmsft

Reply:
Fixed by passing in tokencredentialoptions (which include clientoptions fields such as transport) to the underlying `ClientAssertionCredential` using a similar approach as `EnvironmentCredential`. Thanks @antkmsft for the suggestion.